src/test/java/org/fogbowcloud/manager/requests/api/local/http/ComputeOrdersControllerTest.java

```diff
@@ -48,132 +48,572 @@
+        String COMPUTE_STATUS_ENDPOINT = COMPUTE_ENDPOINT + "/" + ComputeOrdersController.STATUS_ENDPOINT;
+        RequestBuilder requestBuilder = createRequestBuilder(HttpMethod.GET, COMPUTE_STATUS_ENDPOINT, getHttpHeaders(), "");
+        
+        // exercise
         MvcResult result = this.mockMvc.perform(requestBuilder).andReturn();
-
+        
+        // verify
         int expectedStatus = HttpStatus.OK.value();
-        assertEquals(expectedStatus, result.getResponse().getStatus());
-
-        ComputeInstance resultComputeInstance = new Gson().fromJson(result.getResponse().getContentAsString(), ComputeInstance.class);
-        assertTrue(resultComputeInstance != null);
+        Assert.assertEquals(expectedStatus, result.getResponse().getStatus());
+        
+        String expectedResult = "[]";
+        Assert.assertEquals(expectedResult, result.getResponse().getContentAsString());
+        
+        Mockito.verify(this.facade, Mockito.times(1)).getAllInstancesStatus(Mockito.anyString(), Mockito.any(InstanceType.class));
+       
     }
-
+    
+    // test case: Request the list of all computes status when the facade returns a non-empty list. 
+    // Check the response of request and the call of facade for get the computes status.
+    @Test
+    public void testGetAllComputeStatusWhenHasData() throws Exception {
+    	
+    	// set up
+    	final String FAKE_ID_1 = "fake-Id-1";
+    	final String FAKE_ID_2 = "fake-Id-2";
+    	final String FAKE_ID_3 = "fake-Id-3";
+    	
+        InstanceStatus instanceStatus1 = new InstanceStatus(FAKE_ID_1, InstanceState.READY);
+        InstanceStatus instanceStatus2 = new InstanceStatus(FAKE_ID_2, InstanceState.READY);
+        InstanceStatus instanceStatus3 = new InstanceStatus(FAKE_ID_3, InstanceState.READY);
+        
+        List<InstanceStatus> computeStatusList = Arrays.asList(new InstanceStatus[] {instanceStatus1, instanceStatus2, instanceStatus3});
+        Mockito.doReturn(computeStatusList).when(this.facade).getAllInstancesStatus(Mockito.anyString(), Mockito.any(InstanceType.class));
+        
+        String COMPUTE_STATUS_ENDPOINT = COMPUTE_ENDPOINT + "/" + ComputeOrdersController.STATUS_ENDPOINT;
+        RequestBuilder requestBuilder = createRequestBuilder(HttpMethod.GET, COMPUTE_STATUS_ENDPOINT, getHttpHeaders(), "");
+        
+        // exercise
+        MvcResult result = this.mockMvc.perform(requestBuilder).andReturn();
+        
+        // verify
+        int expectedStatus = HttpStatus.OK.value();
+        Assert.assertEquals(expectedStatus, result.getResponse().getStatus());
+        
+        TypeToken<List<InstanceStatus>> token = new TypeToken<List<InstanceStatus>>(){};
+        List<InstanceStatus> resultList = new Gson().fromJson(result.getResponse().getContentAsString(), token.getType());
+        Assert.assertTrue(resultList.size() == 3);
```

Comment:
It seems a Assert.assertEquals(3, resultList.size()) fits better

Reply:
Done